nupic/algorithms/anomaly.py

```diff
@@ -107,6 +124,27 @@ def __init__(self, slidingWindowSize=None, mode=MODE_PURE, binaryAnomalyThreshol
           binaryAnomalyThreshold <= 0.0 ):
       raise ValueError("Anomaly: binaryAnomalyThreshold must be from (0,1) "
                        "or None if disabled.")
+    # switch and setup compute() based on mode
+    self._mode = mode
+    self._likelihood = None
+
+    if mode == Anomaly.MODE_CUSTOM:
+      if callable(customComputeFn):
+        self._computeAnomaly = customComputeFn
+      else:
+        raise ValueError("Anomaly: with mode='custom' a callable must be provided as 'customComputeFn' param")
```

Comment:
Need to run through pylint with Nupic pylint config file, and fix findings. In this case, I suspect that the exception and its text go beyond 80 columns